src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2182,6 +2188,23 @@ private void EnterParameter(ParameterSymbol parameter, TypeWithAnnotations param
                         isDefaultValue: parameter.ExplicitDefaultConstantValue?.IsNull == true);
                 }
             }
+
+            // If we have an initial state, then we're doing the second analysis stage on a constructor
+            // and we shouldn't report assignment warnings related to parameter default values
```

Comment:
Consider adding a `Debug.Assert` for this condition. I would assume the method symbol should be a constructor?

Reply:
I ended up adjusting this logic so that the constructor requirement is verified at runtime.